tools/ed64romconfig.c

```diff
@@ -0,0 +1,197 @@
+#define SAVETYPE_SRAM256K  0x30
+#define SAVETYPE_SRAM768K  0x40
+#define SAVETYPE_FLASHRAM  0x50
+#define SAVETYPE_SRAM1M    0x60
+#define ROMCONFIG_NOT_SET  0xFF
+
+#define CART_ID_OFFSET 0x3C
+#define CART_ID_SIZE   2
+#define VERSION_OFFSET 0x3F
+#define VERSION_SIZE   1
+
+void print_usage(const char * prog_name)
+{
+	fprintf(stderr, "Usage: %s [-r] [-c] [-w <savetype>] <file>\n\n", prog_name);
+	fprintf(stderr, "This program takes a big-endian N64 ROM and sets the header so that\n");
+	fprintf(stderr, "EverDrive64 will respect the declared save type, RTC, and region-free\n");
+	fprintf(stderr, "settings without needing to create a save_db.txt entry for it.\n");
+	fprintf(stderr, "See: https://github.com/krikzz/ED64/blob/master/docs/rom_config_database.md#developer-override\n");
+	fprintf(stderr, "\n");
+	fprintf(stderr, "Command-line flags:\n");
+	fprintf(stderr, "\t-w, --savetype <type>  Declare cartridge save type.\n");
+	fprintf(stderr, "\t-c, --rtc              Declare real-time clock support.\n");
+	fprintf(stderr, "\t-r, --regionfree       Declare region-free ROM.\n");
+	fprintf(stderr, "\n");
+	fprintf(stderr, "Supported cartridge save types:\n");
+	fprintf(stderr, "\tnone        Game does not save or uses Controller Pak.\n");
+	fprintf(stderr, "\teeprom4k    Game saves to 4 kilobit EEPROM.\n");
+	fprintf(stderr, "\teeprom16k   Game saves to 16 kilobit EEPROM.\n");
+	fprintf(stderr, "\tsram256k    Game saves to 256 kilobit SRAM\n");
+	fprintf(stderr, "\tsram768k    Game saves to 768 kilobit SRAM\n");
+	fprintf(stderr, "\tsram1m      Game saves to 1 megabit SRAM\n");
+	fprintf(stderr, "\tflashram    Game saves to 1 megabit FlashRAM\n");
+}
+
+bool check_flag(const char * arg, const char * shortFlag, const char * longFlag)
+{
+	return !strcmp(arg, shortFlag) || !strcmp(arg, longFlag);
+}
+
+/**
+ * Corresponds to ED64 ROM Configuration Database values:
+ * @see https://github.com/krikzz/ED64/blob/master/docs/rom_config_database.md
+ */
+uint8_t parse_save_type(const char * arg)
+{
+	if(!strcmp(arg, "none"))      return SAVETYPE_DISABLED;
+	if(!strcmp(arg, "eeprom4k"))  return SAVETYPE_EEPROM4K;
+	if(!strcmp(arg, "eeprom16k")) return SAVETYPE_EEPROM16K;
+	if(!strcmp(arg, "sram256k"))  return SAVETYPE_SRAM256K;
+	if(!strcmp(arg, "sram768k"))  return SAVETYPE_SRAM768K;
+	if(!strcmp(arg, "flashram"))  return SAVETYPE_FLASHRAM;
+	if(!strcmp(arg, "sram1m"))    return SAVETYPE_SRAM1M;
+	return SAVETYPE_NOT_SET;
+}
+
+/**
+ * Combines save_type, force_rtc, region_free values for use in the header version field:
+ * @see https://github.com/krikzz/ED64/blob/master/docs/rom_config_database.md#developer-override
+ */
+uint8_t rom_configuration(uint8_t save_type, bool force_rtc, bool region_free)
+{
+	if(!force_rtc && !region_free && save_type == SAVETYPE_NOT_SET)
+	{
+		return ROMCONFIG_NOT_SET;
+	}
+	if(save_type == SAVETYPE_NOT_SET)
+	{
+		fprintf(stderr, "WARNING: RTC/Region-Free declared without save type; defaulting to 'none'\n");
```

Comment:
Do we really need this warning? By looking at the command line syntax, I would say it's pretty clear the intended behavior. If I asked for just RTC, I probably meant it.